ci-operator/step-registry/ingress/gatewayapi/conformance-tests/ingress-gatewayapi-conformance-tests-commands.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+echo "Go version: $(go version)"
+cd /go/src/github.com
+mkdir kubernetes-sigs && cd kubernetes-sigs
```

Comment:
Using `&&` here means that the `cd` command won't run if the directory already exists or if the `mkdir` command fails for any other reason:
```text
$ mkdir foo && cd foo
$ pwd
/home/mmasters/foo
$ cd ..
$ mkdir foo && cd foo
mkdir: cannot create directory ‘foo’: File exists
$ pwd
/home/mmasters
$ 
```
`mkdir -p` doesn't return an error if the directory already exists:
```text
$ ls -d foo
foo
$ mkdir -p foo && cd foo
$ pwd
/home/mmasters/foo
$ 
```
So it is better to use `mkdir -p` to avoid an error if the directory already exists (I assume we want to use the existing directory in this case), and use a separate `cd` command so that a failing `mkdir` command causes the script to exit on error:
```suggestion
mkdir -p kubernetes-sigs
cd kubernetes-sigs
```